File: Creational/Abstract Factory/Python/product/suv_parts.py
# product/suv_parts.py
from .auto_body_part import AutoBodyPart

class SUVLeftDoor(AutoBodyPart):
    def get_type(self) -> str:
        return "SUV Left Door"

class SUVRightDoor(AutoBodyPart):
    def get_type(self) -> str:
        return "SUV Right Door"

class SUVHood(AutoBodyPart):
    def get_type(self) -> str:
        return "SUV Hood"

File: Creational/Abstract Factory/TypeScript/src/product/SUVParts.ts
// src/product/SUVParts.ts
import { AutoBodyPart } from './AutoBodyPart';

export class SUVLeftDoor implements AutoBodyPart {
    getType(): string {
        return "SUV Left Door";
    }
}

export class SUVRightDoor implements AutoBodyPart {
    getType(): string {
        return "SUV Right Door";
    }
}

export class SUVHood implements AutoBodyPart {
    getType(): string {
        return "SUV Hood";
    }
}
